services/awsmeter/awsmeter.go

```diff
@@ -0,0 +1,146 @@
+
+	creds := credentials.NewChainCredentials(
+		[]credentials.Provider{
+			&ec2rolecreds.EC2RoleProvider{
+				Client: ec2metadata.New(s),
+			},
+		})
+
+	_, err := creds.Get()
+	if err != nil {
+		mlog.Error("session is invalid", mlog.String("error", err.Error()))
+		return nil, errors.New("cannot obtain credentials")
+	}
+
+	return marketplacemetering.New(session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+	}))), nil
+}
+
+// a report entry is for all metrics
+func (awsm *AwsMeter) GetUserCategoryUsage(dimensions []string, startTime time.Time, endTime time.Time) []*model.AWSMeterReport {
+	reports := make([]*model.AWSMeterReport, 0)
+
+	for _, dimension := range dimensions {
+		var userCount int64
+		var err error
+
+		switch dimension {
+		case model.AWS_METERING_DIMENSION_USAGE_HRS:
+			userCount, err = awsm.store.User().AnalyticsActiveCountForPeriod(model.GetMillisForTime(startTime), model.GetMillisForTime(endTime), model.UserCountOptions{})
+			mlog.Info("GetUserCategoryUsage", mlog.Int64("usercount", userCount), mlog.Err(err), mlog.Bool("bool", err != nil))
+			if err != nil {
+				mlog.Error("Failed to obtain usage data", mlog.String("dimension", dimension), mlog.String("start", startTime.String()), mlog.Int64("count", userCount), mlog.Err(err))
+			}
+		default:
+			mlog.Error("Dimension does not exist!", mlog.String("dimension", dimension))
+			err = errors.New("Dimension does not exist")
+		}
+
+		if err != nil {
+			mlog.Error("Failed to obtain usage.", mlog.String("dimension", dimension), mlog.Err(err))
+			return reports
+		}
+
+		report := &model.AWSMeterReport{
+			Dimension: dimension,
+			Value:     userCount,
+			Timestamp: startTime,
+		}
+
+		reports = append(reports, report)
+	}
+
+	return reports
+}
+
+func (awsm *AwsMeter) ReportUserCategoryUsage(reports []*model.AWSMeterReport) *model.AppError {
+	for _, report := range reports {
+		err := sendReportToMeteringService(awsm.service, report)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func sendReportToMeteringService(ams *AWSMeterService, report *model.AWSMeterReport) *model.AppError {
+	params := &marketplacemetering.MeterUsageInput{
+		DryRun:         aws.Bool(ams.AwsDryRun),
+		ProductCode:    aws.String(ams.AwsProductCode),
+		UsageDimension: aws.String(report.Dimension),
+		UsageQuantity:  aws.Int64(report.Value),
+		Timestamp:      aws.Time(report.Timestamp),
+	}
+
+	resp, err := ams.AwsMeteringSvc.MeterUsage(params)
+	if err != nil {
+		return model.NewAppError("sendReportToMeteringService", "app.system.aws_metering_service.error", nil, err.Error(), http.StatusNotFound)
+	}
+	if resp.MeteringRecordId == nil {
+		return model.NewAppError("sendReportToMeteringService", "app.system.aws_metering_service.error", nil, "", http.StatusNotFound)
+	}
+
+	mlog.Debug("Sent record to AWS metering service", mlog.String("dimension", report.Dimension), mlog.Int64("value", report.Value), mlog.String("timestamp", report.Timestamp.String()))
```

Comment:
I guess we want to keep this. Sounds sensible to have this debug message during development and the Debug log level is ok :)